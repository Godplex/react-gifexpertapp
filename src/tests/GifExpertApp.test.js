import React from 'react';
import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp";
import '@testing-library/jest-dom';
import { GifGrid } from '../components/GifGrid';

describe('Pruebas en GifExpertApp', () => {
    test('Debe mostrar la app correctamente.', () => {
        const wrapper=shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();
    });
    test('Debe de mostrar una lista de categorias ', () => {
        const categories=['One Punch Man','Dragon Ball Z'];
        const wrapper=shallow(<GifExpertApp defaultCategories={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
    
});
