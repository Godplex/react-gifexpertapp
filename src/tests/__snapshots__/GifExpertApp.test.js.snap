// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp Debe de mostrar una lista de categorias  1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch Man"
      key="One Punch Man"
    />
    <GifGrid
      category="Dragon Ball Z"
      key="Dragon Ball Z"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en GifExpertApp Debe mostrar la app correctamente. 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
